<launch>

  <!-- Connect the robot to a keyboard teleop controller -->
  <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" output="screen" respawn="true"/>

  <!-- Run the map server -->
  <arg name="map_file" default="$(find me5413_world)/maps/map.yaml"/>
  <node name="map_server" pkg="map_server" type="map_server" args="$(arg map_file)" />
  
  <!-- Launch the cartographer Localizer, source the cartographer setup.bash file before running! -->
  <node name="cartographer_node" pkg="cartographer_ros"
    type="cartographer_node" args="
        -configuration_directory $(find cartographer_ros)/configuration_files
        -configuration_basename me5413_3d_localization.lua"
    output="screen">
  <remap from="points2" to="/mid/points" />
  <remap from="odom" to="/odometry/filtered" />
  </node>

  <!-- Launch Move Base -->
  <!-- <include file="$(find jackal_navigation)/launch/include/move_base.launch" /> -->
  <include file="$(find me5413_world)/launch/include/move_base_jackal.launch" />

  <!-- Launch Rviz with our settings -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find me5413_world)/rviz/navigation.rviz" output="log" respawn="true"/>

  <node ns="me5413_world" pkg="me5413_world" type="goal_publisher_node" name="goal_publisher_node" output="screen" />

</launch>